src/components/Modal/index.tsx

```diff
@@ -43,31 +66,43 @@ function Modal(props: Props) {
                     styles.modal,
                 )}
             >
-                {heading !== null && (
-                    <div className={_cs(styles.modalHeader, headingClassName)}>
-                        <h2 className={styles.titleContainer}>
-                            {heading}
-                        </h2>
-                        {!closeButtonHidden && (
-                            <div className={styles.actions}>
-                                <Button
-                                    className={styles.closeButton}
-                                    onClick={onClose}
-                                    name="Close"
-                                    variant="action"
-                                >
-                                    <IoClose />
-                                </Button>
-                            </div>
-                        )}
+                {!shouldHideHeader && (
+                    <div className={_cs(styles.modalHeader, headerClassName)}>
+                        <ElementFragments
+                            icons={headerIcons}
+                            actions={(
+                                <>
+                                    {headerActions}
+                                    {!hideCloseButton && (
+                                        <Button
+                                            className={styles.closeButton}
+                                            onClick={onCloseButtonClick}
+                                            name="close-modal"
+                                            variant="action"
+                                        >
+                                            <IoClose />
+                                        </Button>
+                                    )}
+                                </>
+                            )}
+                        >
+                            <h2 className={styles.heading}>
+                                {heading}
+                            </h2>
+                        </ElementFragments>
                     </div>
                 )}
                 <div className={_cs(styles.modalBody, bodyClassName)}>
                     {children}
                 </div>
-                {footer && (
+                {(footer || footerIcons || footerActions) && (
```

Comment:
As we have a hide header option, should we also have a hide footer option?

Reply:
We do not have hide header option, it's just an effect and same goes for the footer